dqc_us_rules/dqc_us_0046.py

```diff
@@ -0,0 +1,100 @@
+# (c) Copyright 2015 - 2017, XBRL US Inc. All rights reserved.
+# See https://xbrl.us/dqc-license for license information.
+# See https://xbrl.us/dqc-patent for patent infringement notice.
+import json
+import os
+from arelle import XbrlConst
+from .util import messages
+
+_CODE_NAME = 'DQC.US.0046'
+_RULE_VERSION = '4.0.0'
+_RULE_INDEX_KEY = '1'
+_NO_FACT_KEY = 'no_fact'
+_CONFIG_JSON_FILE = os.path.join(
+            os.path.dirname(__file__),
+            'resources',
+            'DQC_US_0046',
+            'dqc_0046.json'
+        )
+_EMPTY_LIST = []
+
+
+def _find_errors(val):
+    """
+    Entrypoint for the rule.  Load the config, search for instances of
+    reversed calculation relationships.
+
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :return: No direct return
+    :rtype: None
+    """
+    error_list = []
+    config_json_file = _CONFIG_JSON_FILE
+    calc_children = _load_config(config_json_file)
+    if not calc_children:
+        return  # nothing can be checked
+    # convert children lists into sets for faster "in" function processing
+    calc_children = dict(
+        (key, set(value)) for key, value in calc_children.items()
+    )
```

Comment:
This step seems to be unneeded to me. You convert the list to a set, but later you only check to see if an element is in the set. Im not sure you need a set here since you have control over what these lists look like in the json file. If push comes to shove, you could create the set later on the fly right before the check that way you aren't wasting a bunch of time creating sets for concepts that don't exist in the doc.